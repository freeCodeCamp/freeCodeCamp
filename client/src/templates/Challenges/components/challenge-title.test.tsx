import React from 'react';
import renderer from 'react-test-renderer';

import ChallengeTitle from './challenge-title';

const baseProps = {
  children: 'title text',
  isCompleted: true,
  translationPending: false
};

describe('<ChallengeTitle/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ChallengeTitle {...baseProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
