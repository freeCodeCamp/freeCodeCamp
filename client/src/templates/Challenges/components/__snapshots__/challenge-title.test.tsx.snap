// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChallengeTitle/> renders correctly 1`] = `
<div
  className="challenge-title-wrap"
>
  <div
    className="challenge-title"
  >
    <h1
      data-playwright-test-label="challenge-title"
      id="content-start"
    >
      title text
    </h1>
    <svg
      aria-label="icons.passed"
      height="15"
      viewBox="0 0 200 200"
      width="15"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        aria-hidden="true"
      >
        <title>
          icons.passed
        </title>
        <circle
          cx="100"
          cy="99"
          fill="var(--primary-color)"
          r="95"
          stroke="var(--primary-color)"
          strokeDasharray="null"
        />
        <rect
          fill="var(--primary-background)"
          height="30"
          stroke="var(--primary-background)"
          strokeDasharray="null"
          transform="rotate(-45, 120, 106.321)"
          width="128.85878"
          x="55.57059"
          y="91.32089"
        />
        <rect
          fill="var(--primary-background)"
          height="30"
          stroke="var(--primary-background)"
          strokeDasharray="null"
          transform="rotate(45, 66.75, 123.75)"
          width="80.66548"
          x="26.41726"
          y="108.75"
        />
      </g>
    </svg>
  </div>
</div>
`;
